### PR TITLE
Solves truncated proxy results, fixes 190

### PR DESCRIPTION
**Pull request by phpmaps**
This is hopefully a clean pull request.

Made sure that the proxy captures the incoming request headers from the client and uses them when making the cURL request.

Next the PHP proxy reads the cURL response and collects all the headers. These headers are then sent to the client along with the response.

This fixes reports of truncated responses. This was b/ the Content-Length property was incorrect. The proxy now calculates Content-Length and sets the proper Content-Length property.

### CODE REVIEW

**Comment by jgravois** [comment]
this PR is looking good :+1:

**Comment by bsvensson** [comment]
thanks @jgravois @phpmaps - I'll doublecheck locally then merge

File: PHP/proxy.php [hunk 1]
```diff
@@ -591,28 +719,27 @@ public function canProcessRequest()
...
-
-                if(is_string($s['matchAll'])){
-
-                    $mustmatch = strtolower($s['matchAll']);
+                $serverUrl['url'] = $this->sanitizeUrl($serverUrl['url']); //Do all the URL cleanups and checks at once
+                
+                if(is_string($serverUrl['matchall'])){
 
-                    $s['matchAll'] = $mustmatch;
+                    $serverUrl['matchAll'] = strtolower($serverUrl['matchall']);
```
Comment (bsvensson) [inline PHP/proxy.php]: @phpmaps - should that not be lowercase matchall instead of matchAll with Capital A in ` $serverUrl['matchAll'] = strtolower...`

**Comment by phpmaps** [comment]
@bsvensson  correct.  The best way to access configuration values is using lowercase keys.  When the PHP proxy reads the config file - it converts all config properties to lowercase and creates a series of arrays using those lowercase values as keys.

Here is an output, I captured to show how it truly is lowercase.

```
array (size=1)
  'serverurl' => 
    array (size=1)
      0 => 
        array (size=6)
          'url' => string 'http://sampleserver6.arcgisonline.com/arcgis/rest/services/SaveTheBay/FeatureServer' (length=83)
          'username' => string 'user2' (length=5)
          'password' => string 'user2' (length=5)
          'ratelimit' => string '120' (length=3)
          'ratelimitperiod' => string '60' (length=2)
          'matchall' => string 'true' (length=4)

```
 In regards to fixing #189 my thoughts are we reopen #189 and discuss. I suspect the problem is in the actual matching logic. Should you guys merge this pull request... it actually sets things back to lowercase.  Let me know.

**Comment by bsvensson** [comment]
OK, I'm fine re-fixing #189 with the better solution (only lowercase one "A"). I'll take care of that once this is merged in.

**Pull request merged**